ibm_mq/README.md

```diff
@@ -2,31 +2,112 @@
 
 ## Overview
 
-This check monitors [Ibm_mq][1].
+This check monitors [IBM MQ][1].
 
 ## Setup
 
 ### Installation
 
-The Ibm_mq check is included in the [Datadog Agent][2] package. However, in order for it to work it needs to have an IBM MQ client installed on the box.
+The IBM MQ check is included in the [Datadog Agent][2] package.
+
+In order to use the IBM MQ check, you need to install the [IBM MQ Client][3], unless the IBM MQ server is already installed on the box. Take note of where you installed it.
+
+If you are using Linux, after the MQ Client is installed ensure the runtime linker can find the libraries. For example, using ldconfig:
+
+```
+# Put the library location in an ld configuration file.
+
+sudo sh -c "echo /opt/mqm/lib64 > \
+/etc/ld.so.conf.d/mqm64.conf"
+sudo sh -c "echo /opt/mqm/lib > \
+/etc/ld.so.conf.d/mqm.conf"
+
+# Update the bindings.
+
+sudo ldconfig
+```
+
+Alternately, update your LD_LIBRARY_PATH to include the location of the libraries. For example:
+
+```
+export LD_LIBRARY_PATH=/opt/mqm/lib64:/opt/mqm/lib:$LD_LIBRARY_PATH
+```
+
+*Note*: Agent 6 uses Upstart or systemd to orchestrate the datadog-agent service. Environment variables may need to be added to the service configuration files at the default locations of /etc/init/datadog-agent.conf (Upstart) or /lib/systemd/system/datadog-agent.service (systemd). See documentation on Upstart or systemd for more information on how to configure these settings.
+
+Here's an example of the configuration that's used for Systemd:
+
+```yaml
+[Unit]
+Description="Datadog Agent"
+After=network.target
+Wants=datadog-agent-trace.service datadog-agent-process.service
+StartLimitIntervalSec=10
+StartLimitBurst=5
+
+[Service]
+Type=simple
+PIDFile=/opt/datadog-agent/run/agent.pid
+Environment="LD_LIBRARY_PATH=/opt/mqm/lib64:/opt/mqm/lib"
```

Comment:
Should this also have `:$LD_LIBRARY_PATH` appended?

Reply:
Maybe? It's unclear to me what that would do since systemd units aren't loaded from within a shell it isn't super clear to me that it'll be interpolated in the way we think it will

Reply:
Maybe? It's unclear to me what that would do since systemd units aren't loaded from within a shell it isn't super clear to me that it'll be interpolated in the way we think it will